webpack/ForemanTasks/Components/TasksTable/Components/ConfirmModal/ConfirmModal.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { startCase } from 'lodash';
+import { sprintf, translate as __ } from 'foremanReact/common/I18n';
+import { Button } from 'patternfly-react';
+import ForemanModal from 'foremanReact/components/ForemanModal';
+import { useForemanModal } from 'foremanReact/components/ForemanModal/ForemanModalHooks';
+import { CANCEL_MODAL, RESUME_MODAL } from '../../TasksTableConstants';
+
+export const ConfirmModal = ({
+  actionText,
+  actionState,
+  tasksActions,
+  actionType,
+  selectedRowsLen,
+  id,
+}) => {
+  const { setModalClosed } = useForemanModal({
+    id,
+  });
+  if ([CANCEL_MODAL, RESUME_MODAL].includes(actionType)) {
+    selectedRowsLen = 1;
+  }
+  return (
+    <ForemanModal
+      title={sprintf(__('%s Selected Tasks'), startCase(actionText))}
+      id={id}
+    >
+      {sprintf(
+        __(
+          `This will %(action)s %(number)s task(s), putting them in the %(state)s state. Are you sure?`
+        ),
+        { action: actionText, number: selectedRowsLen, state: actionState }
```

Comment:
this two `sprintf` can be moved to the parent too,
and consumed here by `title, body` props for example.

Reply:
Why? this is not intended to be a generic component, but a component for the tasks table to divide the logic in the tasksTablePage